src/boards/rainbow_esp.cpp

```diff
@@ -87,11 +87,13 @@ BrokeStudioFirmware::BrokeStudioFirmware() {
 	char const* hostname = ::getenv("RAINBOW_SERVER_ADDR");
 	if (hostname == nullptr) hostname = "";
 	this->server_settings_address = hostname;
+	this->default_server_settings_address = hostname;
 
 	char const* port_cstr = ::getenv("RAINBOW_SERVER_PORT");
-	if (port_cstr == nullptr) port_cstr = "3000";
+	if (port_cstr == nullptr) port_cstr = "0";
 	std::istringstream port_iss(port_cstr);
 	port_iss >> this->server_settings_port;
+	port_iss >> this->default_server_settings_port;
```

Comment:
It won't work (you are extracting two integers from a string containing only one.)

Replace with `this->default_server_settings_port = this->server_settings_port;`